Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -160,6 +153,269 @@ theorem support_eq_empty {p} : p.support = ∅ ↔ (p : SkewMonoidAlgebra k G) =
+
+theorem single_zero (a : G) : (single a 0 : SkewMonoidAlgebra k G) = 0 := by
+  rw [ofFinsupp_eq_zero, single, Finsupp.single_zero]
+
+theorem single_eq_zero {a : G} {b : k} : single a b = 0 ↔ b = 0 := by
+  rw [ofFinsupp_eq_zero, single, Finsupp.single_eq_zero]
+
+/-- Group isomorphism between `SkewMonoidAlgebra k G` and `G →₀ k`. This is an
+implementation detail, but it can be useful to transfer results from `Finsupp`
+to `SkewMonoidAlgebra`. -/
+@[simps apply symm_apply]
+def toFinsuppAddEquiv : SkewMonoidAlgebra k G ≃+ (G →₀ k) where
+  toFun        := toFinsupp
+  invFun       := ofFinsupp
+  left_inv     := fun ⟨_p⟩ => rfl
+  right_inv _p := rfl
+  map_add'     := toFinsupp_add
+
+theorem smul_single {S} [SMulZeroClass S k] (s : S) (a : G) (b : k) :
+    s • single a b = single a (s • b) :=
+  toFinsupp_injective <| by simp;
+
+theorem single_injective (a : G) : Function.Injective (single a : k → SkewMonoidAlgebra k G) :=
+  toFinsuppAddEquiv.symm.injective.comp (Finsupp.single_injective a)
+
+theorem _root_.IsSMulRegular.skewMonoidAlgebra_iff {S : Type*} [Monoid S] [DistribMulAction S k]
+    {a : S} [inst : Nonempty G]: IsSMulRegular k a ↔ IsSMulRegular (SkewMonoidAlgebra k G) a := by
+  refine ⟨IsSMulRegular.skewMonoidAlgebra, fun ha b₁ b₂ inj ↦ ?_⟩
+  rw [← (single_injective _).eq_iff, ← smul_single, ← smul_single] at inj
+  exact single_injective (Classical.choice inst) (ha inj)
+
+end Single
+
+section One
+
+variable [One G] [One k]
+
+/-- The unit of the multiplication is `single 1 1`, i.e. the function that is `1` at `1` and
+  zero elsewhere. -/
+instance one : One (SkewMonoidAlgebra k G) := ⟨single 1 1⟩
+
+theorem ofFinsupp_one : (⟨Finsupp.single 1 1⟩ : SkewMonoidAlgebra k G) = 1 := rfl
+
+@[simp]
+theorem toFinsupp_one : (1 : SkewMonoidAlgebra k G).toFinsupp = Finsupp.single 1 1 := rfl
+
+@[simp]
+theorem toFinsupp_eq_one {a : SkewMonoidAlgebra k G} :
+    a.toFinsupp = Finsupp.single 1 1 ↔ a = 1 := by
+  rw [← toFinsupp_one, toFinsupp_inj]
+
+@[simp]
+theorem ofFinsupp_eq_one {a} :
+    (⟨a⟩ : SkewMonoidAlgebra k G) = 1 ↔ a = Finsupp.single 1 1 := by
+  rw [← ofFinsupp_one, ofFinsupp_inj]
+
+theorem single_one_one  : single 1 (1 : k) = 1 := rfl
+
+theorem one_def : (1 : SkewMonoidAlgebra k G) = single 1 1 := rfl
+
+@[simp]
+theorem coeff_one_one : coeff (1 : SkewMonoidAlgebra k G) 1 = 1 := by
+  simp only [coeff, toFinsupp_single, Finsupp.single_eq_same]
+
+theorem coeff_one {a : G} [Decidable (a = 1)] :
+    (1 : SkewMonoidAlgebra k G).coeff a = if a = 1 then 1 else 0 := by
+  classical
+  simp_rw [eq_comm (a := (a : G)) (b := (1 : G))]
+  simpa using coeff_single
+
+end One
+
+section sum
+
+instance [DecidableEq G] [DecidableEq k] : DecidableEq (SkewMonoidAlgebra k G) :=
+  Equiv.decidableEq toFinsuppAddEquiv.toEquiv
+
+/-- `sum f g` is the sum of `g a (f.coeff a)` over the support of `f`. -/
+def sum {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) : N :=
+  ∑ a ∈ f.support, g a (f.coeff a)
```

Comment:
I think `f.toFinsupp.sum` would be a more convenient definition. You can leave `sum_def` as currently written.